Workflow for isobaric analysis (untested)

oliveralka opened the pull request:
Reduced rt setting in IDMapper 
TMT6-plex (K,N-term) in MSGFPlus
The correction Matrix for the TMT experiment has to be set by the user!

jpfeuffer commented:
Very nice. I didn't have a look yet (will today).
But:
- Did you make the input files accept relative paths inside the workspace?
- Did you add a description of the workflow in the workflow settings for easy upload to the KNIME Hub?

jpfeuffer commented:
I will write a TODO list for PRs here.

jpfeuffer commented:
Ah this was just a change in one of the WFs but still, maybe you could add the abovementioned points. We should do it for every WF here anyway soon.

oliveralka commented:
> Did you make the input files accept relative paths inside the workspace? 

Should we ship the input data with the workflow? 

> Did you add a description of the workflow in the workflow settings for easy upload to the KNIME Hub?

Can you provide a reference / or example on how to do that?

jpfeuffer commented:
No, otherwise it gets too big. Data goes here (assume the workspace root is this folder, i.e. specify paths relative to that root):
https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Tutorials/Data/

Instructions for description is here:
https://docs.google.com/document/d/1uwVCz-4LH7VUzhcsE5FVyrBbRmzDnGQvdFF2ZsA8BYU/edit#heading=h.3hm698qmdoi3

jpfeuffer commented:
The workflows have to run from the commandline. You can check by running the shell script executeBatch with arguments:
1) knime executable
2) workspace root

oliveralka commented:
There are some errors in the workflow which need to be fixed first - I will update the PR, when it is ready.

oliveralka commented:
We have to check if the IDMapper is switched to "precursor" see https://github.com/OpenMS/OpenMS/issues/4216

pavel-shliaha commented:
Is the pipeline available for MS3 SPS?

oliveralka commented:
The IsobaricAnalyzer should use MS3 automatically if possible (@cbielow).

oliveralka commented:
@timosachsenberg @jpfeuffer 
Added a rought draft for integration into the tutorial. Updated the workflow using epifany - runable - but I still have to check the results. 
It would be great if you could take a look and make some comments and suggestions, also in terms of explanation of the experimental design for TMT. 

Julianus, could you have a peek at the parameters used for epifany?

pavel-shliaha commented:
@oliveralka could you please point to where the tutorial for MS3 TMT is and the actual workflow that you recommend?

pavel-shliaha commented:
@oliveralka could you please point me to the tutorial and also to the KNIME workflow?

pavel-shliaha commented:
@oliveralka. I have found the workflow, but I cannot find the example files (mzMLs and also the MSstatsTMT table). Could you perhaps give a direct link to it?

oliveralka commented:
@pavel-shliaha  The small example can be found here:

pavel-shliaha commented:
@oliveralka. I downloaded the newest KNIME and OpenMS, the dataset and the database and set file locations on my computer. MSGFPlusAdapter and IsobaricAnalyzer nodes failed. Do they work in your hands?

oliveralka commented:
@pavel-shliaha 
Everything works in my case: 
KNIME 4.1.1 
OpenMS 2.5.0.20200201806

What are the error messages? Can you the tools in debug mode?

pavel-shliaha commented:
@oliveralka. I have

KNIME 4.1.2 
OpenMS 2.5.0.202002241222

I installed the KNIME and workflow on two separate computers and both of them failed (I downloaded the data from the repository you suggested and set up the files to the corresponding nodes). I used the workflow you suggested. How do I run the software in debug mode (cannot find it online)? I can also give access to you for my computer if you want to have a look.

pavel-shliaha commented:
@oliveralka. I tried to execute the workflow on two separate computers. On both of them:

KNIME is 4.1.2
OpenMS is 2.5.0.202002241222

 How can I run the software in debug mode (googled the question but could not find an answer)? Also if you are interested I can provide access to my computer so you can have a look.

jpfeuffer commented:
What does the Console log in the lower right of KNIME say?

pavel-shliaha commented:
@jpfeuffer it says I might be missing some of the requirements (trying to isntall them now). I appologise for this but I used the website https://www.openms.de/download/knime-plugin/ and it does not mention the prerequisites. I will now install proteowizard as well

jpfeuffer commented:
ok we will update the documentation. But also check the other tab of the lower right view called
"Console". And if you need to, you can also check single tools by right-clicking the node and selecting View Standard Output/Error.

pavel-shliaha commented:
When I load the workflow I get the following message: 

<img width="960" alt="error" src="https://user-images.githubusercontent.com/1895805/76268664-a903b480-6266-11ea-96ea-0960337c4511.PNG">

jpfeuffer commented:
that can usually be ignored.

pavel-shliaha commented:
I installed the prerequisites and the prerequisites for visual studio under this link 

https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads

but it did not help

jpfeuffer commented:
As long as it just shows "... FileConverter requirements" this can also be ignored. You just might not be able to use Raw file conversion inside KNIME. We just check registry keys here. Sometimes it does not work correctly.

pavel-shliaha commented:
it still fails to execute the workflow

jpfeuffer commented:
I thought so. Without the requirements no OpenMS node would work. That is why you need to check the Console and the Standard Output/Error of the nodes for info now.

jpfeuffer commented:
Just a random guess: did you configure the Input database in the ID Metanode?

pavel-shliaha commented:
I have installed:

1. KNIME
2. OpenMS-2.5.0-Win64.exe from https://www.openms.de/download/openms-binaries
3. OpenMS extension through KNIME
4. OpenMS-2.5-prerequisites-installer.exe 

the workflow still does not run. Should I install anything else?

pavel-shliaha commented:
I have reimported the workflow and it seems to be working

jpfeuffer commented:
@oliveralka did you check: https://github.com/OpenMS/Tutorials/pull/166#issuecomment-529812404

oliveralka commented:
@jpfeuffer Yes, checked some time ago - is set to precursor.

jpfeuffer commented on Handout/isobaric.tex:
```diff
@@ -0,0 +1,114 @@
+
+\newpage
+\section{Isobaric analysis}
+In the last chapters, we identified and quantified peptides in a label-free experiment. In this section, we would like to introduce a possible workflow for the analysis of isobaric data. We have prepared and copied this workflow to the USB sticks. Please import \directory{Workflows > IIdentification\_quantification\_isobaric\_MSstatsTMT } into KNIME via the menu entry \menu{File > Import KNIME wo ...
```
Spelling: IIdentification

jpfeuffer replied:
Mention somehow that this can be used for iTraq as well.

jpfeuffer commented on Handout/isobaric.tex:
```diff
@@ -0,0 +1,114 @@
+\subsection{Experimental Design (TMT)}
+The experimental design for a TMT experiment is similar to a LFQ experiment. The difference is in the available channels and possible Mixtures. 
+
+\begin{table}[!ht]
```
Table is way to wide: https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Tutorials/Handout/oliveralka-patch-2/handout.pdf

oliveralka replied:
Split into two tables with 0.95 textwidth.

jpfeuffer commented on Handout/isobaric.tex:
```diff
@@ -0,0 +1,114 @@
+The workflow has three input nodes, the first for the experimental design to allow for MSstatsTMT compatible export (.tsv). The second for the .mzML files with the centroided spectra data from the isobaric labeling experiment and the last one for the .fasta database used for identification. The quantification (A) is performed using the \KNIMENODE{IsobaricAnalzyer}. The tool is able to extract and ...
+
+The workflow is performed peptide level (B, D, F, H), were the posterior error probability (PEP) estimation and FDR filtering is performed on PSM level for each file individually (B). Afterwards the identification (PSM) and quantiative information is combined using the \KNIMENODE{IDMapper}. After the processing of all available files, the intermediate results are aggregated (D) and can be exporte ...
+compatible version. Here, the R package MSstatsTMT can be used for further processing. \\
```
Add a TODO that protein inference will be added in a later version of the tutorial.

pavel-shliaha commented:
> Looks good. Minor changes.
> Then we can go over the workflow together via Skype.

Dear Jennifer and Oliver, could you please add me to the skype discussion as well? I use a lot of MS3 SPS and I think I couldprovide some valuable input t discussion, as well as clarify some questions for myself. I wanted to ask about whether certain functionalities are implemented in openMS. In particular:

1) Could yo clarify why IDConflictResolver is used with isobaric mass tagging?
2) I understand that OpenMS workflow suggests using MSStats node for protein inference, but I actually wanted to export the ids with quant values and then look through the table myself. Is there a node that will convert IDConflictResolver output consensusXML to a csv output where the quantitation values are one line for PSM (wide format) or one line per channel (long format). I tried TextExporter, but it does not seem to generate this output
3) is there a node to estimate precursor contamination?
4) is there a node to estimate which of the fragments selected for MS3 originate from MS2 fragments which have been identified as true fragments from the identified peptide?

jpfeuffer commented:
Hi Pavel,

sorry, the meeting is right now. But you will be in the data clinic tomorrow right?

1) It only makes sure that the top hit for the corresponding MS2 spectrum is retained (and not more). You could probably achieve the same with an IDFilter after the search engine.
2) The MSstats node just rearranges the tables to be compatible with MSstats. It does not perform any inference.
TextExporter should export one line per PSM in the "FEATURE" section. The intensities should be in the columns intensity_0 - intensity_9 (or similar). An alternative would be the MzTabExporter (but I did not double-check the completeness of the output for TMT yet).
3) Yes, if you export meta values in the TextExporter, there should be a column with a precursor_purity score.
4) No, but that is a good idea.

jpfeuffer commented:
@pavel-shliaha Regarding 4): How do you get the information about the isolated ions in the MS2. I have an SPS mzML here and all the MS3 spectra only report a single selected precursor ion:
```
              <selectedIonList count="1">
                <selectedIon>
                  ...
```
 Not sure if this is a bug in the mzML converters (e.g. proteowizard) or if this information is just not available in the raw file.

pavel-shliaha commented:
I am looking at the MSGFPlusAdapter and it does not have the TMTPro modification implemented. Could you please have a look?

jpfeuffer commented:
Hi, which Unimod accession is this?

pavel-shliaha commented:
accession number is 2016

oliveralka commented:
I added the data here: https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Tutorials/Data/isobaric_MSV000084264/ 

@jpfeuffer The path to the data has still to be added to the document, but please let me know what you think of the current version. 

[handout_isobaric_20200729.pdf](https://github.com/OpenMS/Tutorials/files/4994202/handout_isobaric_20200729.pdf)
